File: reb_4_5_4/matlab/reb_4_5_4.m
function reb_4_5_4

    % an Arrhenius parameters function named Arrhenius_parameters was
    % written and saved in a folder in the Matlab path

    % function that performs the analysis
	function perform_the_analysis()
    
        % define the ideal gas constant
        R = 8.314E-3; % kJ/mol/K
    
        % read the data file
        data_file = "../reb_4_5_4_data.csv";
        data_table = readtable(data_file,'VariableNamingRule','preserve');
        T = table2array(data_table(:,2)); % °C
        k = table2array(data_table(:,3)); % L/mol/min
    
        % create a temperature data set in absolute units
        T = T + 273.15;
    
        % calculate the Arrhenius parameters and statistics
        [k0, k0_ci, E, E_ci, r_squared] = Arrhenius_parameters(k,T,R);

        % generate, show, and save a model plot 
        k_pred = k0*exp(-E/R./T);
        figure
        hold on
        plot(log(k),1./T,'or','MarkerFaceColor','r','MarkerSize',10)
        plot(log(k_pred),1./T,'k',LineWidth=2)
        hold off
        set(gca, 'FontSize', 14);
        xlabel('1/T (K^-^1)','FontSize', 14)
        ylabel('k (L mol^-^1 min^-^1)','FontSize', 14)
        saveas(gcf,"Arrhenius_plot.png")
    
        % show the results
        disp(' ')
        disp(['k0 = ',num2str(k0,3),' [',num2str(k0_ci(1),3),...
            ', ',num2str(k0_ci(2),3),'] (95%CI) L/mol/min'])
        disp(['E = ',num2str(E,3),' [',num2str(E_ci(1),3),...
            ', ',num2str(E_ci(2),3),'] (95%CI) kJ/mol'])
        disp(['R_squared = ',num2str(r_squared,3)])
    
        % tabulate and save the results
        item = ["k0";"k0_lower_limit";"k0_upper_limit"
            "E";"E_lower_limit";"E_upper_limit";"R_squared"];
        value = round([k0;k0_ci(1);k0_ci(2);E;E_ci(1);
            E_ci(2);r_squared],3,'significant');
        units = ["L/mol/min";"L/mol/min";"L/mol/min";"kJ/mol";"kJ/mol"
            "kJ/mol";""];
        results_table = table(item,value,units);
        writetable(results_table,'results.csv');
    end

    % perform the analysis
    perform_the_analysis()
end

File: reb_4_5_4/python/reb_4_5_4.py
from score_utils import Arrhenius_parameters
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import math

# an Arrhenius parameters function named Arrhenius_parameters was
# written and added to the rebutils library

# function that performs the calculations
def perform_the_calculations():

    # define the ideal gas constant
    R = 8.314E-3 # kJ/mol/K

    # read the data file and convert the temperature to K
    df = pd.read_csv('reb_4_5_4/reb_4_5_4_data.csv')
    df.columns = ['i','T','k']
    T = df['T'].to_numpy() + 273.15
    k = df['k'].to_numpy()

    # calculate the Arrhenius parameters and statistics
    k0, k0_ci, E, E_ci, r_squared = Arrhenius_parameters(k,T,R)

    # generate and save a model plot
    y_pred = k0*np.exp(-E/R/T)
    plt.figure()
    plt.semilogy(1/T,k,color='r',marker='o', ls='none')
    plt.semilogy(1/T,y_pred,color='k')
    plt.xlabel('T$^{-1}$ (K$^{-1}$)')
    plt.ylabel('k (L mol$^{-1}$ min$^{-1}$)')
    # save and show the figure
    plt.savefig('reb_4_5_4/python/Arrhenius_plot.png')
    plt.show()

    # show the results
    print(' ')
    print(f'k0: {k0:.3g} L/mol/min, 95% CI [{k0_ci[0]:.3g}, {k0_ci[1]:.3g}]')
    print(f'E: {E:.3g} kJ/mol, 95% CI [{E_ci[0]:.3g}, {E_ci[1]:.3g}]')
    print(f'R-squared: {r_squared:.3g}')
    print(' ')

    # tabulate and save the results
    data = [['k0', f'{k0:.3g}', 'L mol^-1^ min^-1^'],
        ['k0_lower_limit', f'{k0_ci[0]:.3g}', 'L mol^-1^ min^-1^'],
        ['k0_upper_limit', f'{k0_ci[1]:.3g}', 'L mol^-1^ min^-1^'],
        ['E', f'{E:.3g}', 'kJ mol^-1^'],
        ['E_lower_limit', f'{E_ci[0]:.3g}', 'kJ mol^-1^'],
        ['E_upper_limit', f'{E_ci[1]:.3g}', 'kJ mol^-1^'],
        ['R_squared', f'{r_squared:.3g}', '']]
    result = pd.DataFrame(data, columns=['item','value','units'])
    result.to_csv("reb_4_5_4/python/results.csv", index=False)

    return

if __name__=="__main__":
    perform_the_calculations()
